### PR TITLE
GNA add SplitConvolution transformation unit tests

### PR DESCRIPTION
**Pull request by evkotov**
### Details:
- add SplitConvolutiontransformation unit tests
- fix SplitConvolutiontransformation transformation with copying runtime info

### Tickets:
 - 57583

### CODE REVIEW

File: inference-engine/tests/unit/gna/ngraph/transformations/gna_split_convolution_with_large_buffer_size.cpp [hunk 1]
```diff
@@ -0,0 +1,597 @@
...
+
+namespace testing {
+
+// use constexpr uint32_t bufferMaxSize = 65528 from gna_limitations
+
+TEST(TransformationTests, SplitConvolutionTest) {
+    std::shared_ptr<ngraph::Function> func(nullptr), reference_func(nullptr);
+
+    {
+        auto input_params = std::make_shared<ngraph::opset7::Parameter>(ngraph::element::f32,
```
Comment (elilobanova) [inline inference-engine/tests/unit/gna/ngraph/transformations/gna_split_convolution_with_large_buffer_size.cpp:23]: is it possible here to create some generic function which creates a subgraph with 1 convolution or with several convolutions and just gets shapes and a number of convolutions?

File: inference-engine/tests/unit/gna/ngraph/transformations/gna_split_convolution_with_large_buffer_size.cpp [hunk 2]
```diff
@@ -0,0 +1,597 @@
...
+        auto result = std::make_shared<ngraph::opset7::Result>(concat);
+        reference_func = std::make_shared<ngraph::Function>(ngraph::ResultVector{result},
+                                              ngraph::ParameterVector{input_params});
+    }
+    const FunctionsComparator func_comparator = FunctionsComparator::with_default().enable(FunctionsComparator::ATTRIBUTES);
+    const FunctionsComparator::Result result = func_comparator(func, reference_func);
+    ASSERT_TRUE(result.valid);
+}
+
+TEST(TransformationTests, SplitConvolutionWithFqVariant1TestSmallSize) {
```
Comment (elilobanova) [inline inference-engine/tests/unit/gna/ngraph/transformations/gna_split_convolution_with_large_buffer_size.cpp:360]: what is a difference between Variant1 and Variant2? Can these tests be renamed to contain some information regarding this difference?

File: inference-engine/tests/unit/gna/ngraph/transformations/gna_split_convolution_with_large_buffer_size.cpp [hunk 3]
```diff
@@ -0,0 +1,352 @@
...
+    std::shared_ptr<ngraph::opset7::Parameter> input_node;
+    ngraph::OutputVector output_nodes;
+};
+
+SubGraph createSubGraphSolid(const ngraph::Shape& input_shape, const ngraph::Shape& kernel_shape) {
+    SubGraph sub_graph;
+
+    sub_graph.input_node = std::make_shared<ngraph::opset7::Parameter>(ngraph::element::f32,
+                                                                       input_shape);
+    auto kernel = ngraph::opset1::Constant::create(ngraph::element::f32,
```
Comment (nkogteva) [inline inference-engine/tests/unit/gna/ngraph/transformations/gna_split_convolution_with_large_buffer_size.cpp:30]: oset7

File: inference-engine/tests/unit/gna/ngraph/transformations/gna_split_convolution_with_large_buffer_size.cpp [hunk 4]
```diff
@@ -0,0 +1,352 @@
...
+    auto split_node_c1 = ngraph::opset7::Constant::create(ngraph::element::i64, ngraph::Shape({1}), std::vector<int64_t>{3});
+    auto split_node_c2 = ngraph::opset7::Constant::create(ngraph::element::i64, ngraph::Shape({5}), {960, 960, 960, 960, 256});
+    auto split_node = std::make_shared<ngraph::opset7::VariadicSplit>(sub_graph.input_node,
+                                                                      split_node_c1,
+                                                                      split_node_c2);
+
+    auto kernel = ngraph::opset1::Constant::create(ngraph::element::f32,
+                                                   ngraph::Shape{1, 64, 1, 1}, {1});
+
+    for (int i = 0; i < 5; ++i) {
```
Comment (nkogteva) [inline inference-engine/tests/unit/gna/ngraph/transformations/gna_split_convolution_with_large_buffer_size.cpp:59]: It does not seem to be universal. The number of splits depends on the input shape. I think this function should be parametrized as well.

**Review by nkogteva — APPROVED** [review]

File: inference-engine/tests/unit/gna/ngraph/transformations/gna_split_convolution_with_large_buffer_size.cpp [hunk 5]
```diff
@@ -0,0 +1,330 @@
...
+    return createUnique<DecorT>(std::move(build_decorator));
+}
+
+template<typename DecorT, typename... DecorTs>
+auto createBuildDecorator(const ngraph::Shape& input_shape,
+                                             const ngraph::Shape& kernel_shape)
+                                                -> typename std::enable_if<(sizeof...(DecorTs) > 0), CreateGraphDecoratorPtr>::type {
+    CreateGraphDecoratorPtr build_decorator = createBuildDecorator<DecorTs...>(input_shape, kernel_shape);
+    return createUnique<DecorT>(std::move(build_decorator));
+}
```
Comment (pelszkow) [inline inference-engine/tests/unit/gna/ngraph/transformations/gna_split_convolution_with_large_buffer_size.cpp:244]: Consider to split template conditions and return types: https://godbolt.org/z/conbefe6s

File: inference-engine/tests/unit/gna/ngraph/transformations/gna_split_convolution_with_large_buffer_size.cpp [hunk 6]
```diff
@@ -0,0 +1,330 @@
...
+using CreateGraphDecoratorPtr = std::unique_ptr<CreateGraphDecorator>;
+
+class CreateAppendableGraphDecorator : public CreateGraphDecorator {
+public:
+    CreateAppendableGraphDecorator(std::unique_ptr<CreateGraphDecorator> prev = nullptr) :
+        CreateGraphDecorator(std::move(prev)) {}
+protected:
+    void updateGraph(Graph& graph) override {
+        ngraph::OutputVector new_graph_output;
+        for (auto& node : graph.output_nodes) {
```
Comment (pelszkow) [inline inference-engine/tests/unit/gna/ngraph/transformations/gna_split_convolution_with_large_buffer_size.cpp:66]: `auto&` is potential dangling. Suggest `auto&&` instead.

**Review by elilobanova — APPROVED** [review]

**Pull request merged**